python/src/spark_rapids_ml/umap.py

```diff
@@ -1201,36 +1247,47 @@ def _pre_process_data(
         return select_cols, multi_col_names, dimension, feature_type
 
 
-class UMAPModel(_CumlModel, UMAPClass, _UMAPCumlParams):
+class UMAPModel(_CumlModelWithColumns, UMAPClass, _UMAPCumlParams):
     def __init__(
         self,
-        embedding_: List[pyspark.broadcast.Broadcast],
-        raw_data_: List[pyspark.broadcast.Broadcast],
+        embedding_: np.ndarray,
+        raw_data_: Union[
+            np.ndarray,
+            scipy.sparse.csr_matrix,
+        ],
+        sparse_fit: bool,
         n_cols: int,
         dtype: str,
     ) -> None:
         super(UMAPModel, self).__init__(
             embedding_=embedding_,
             raw_data_=raw_data_,
+            sparse_fit=sparse_fit,
             n_cols=n_cols,
             dtype=dtype,
         )
         self.embedding_ = embedding_
         self.raw_data_ = raw_data_
+        self._sparse_fit = sparse_fit  # If true, raw data is a sparse CSR matrix
+        self.BROADCAST_LIMIT = 8 << 30  # Spark broadcast limit: 8GiB
```

Comment:
Would this limit be affected by spark configuration parameters such as spark.driver.maxResultSize? Maybe create a ticket if we this needs to be investigated.